### PR TITLE
Fixed exception handling in TransmitAsync

### PR DESCRIPTION
**Pull request by milopezc**
Fixed exception handling in TransmitAsync.

### CODE REVIEW

File: src/StreamJsonRpc/Resources.resx [hunk 1]
```diff
@@ -281,4 +281,8 @@
   <data name="ParameterObjectsNotSupportedInJsonRpc10" xml:space="preserve">
     <value>JSON-RPC 1.0 does not support named arguments (parameters passed within an object). Use positional arguments (parameter arrays) or set protocol version to 2.0.</value>
   </data>
+  <data name="ErrorWritingJsonRpcMessage" xml:space="preserve">
+    <value>Error writing JSON RPC Result: {0}: {1}</value>
```
Comment (AArnott) [inline src/StreamJsonRpc/Resources.resx]: "Result" here should probably be "message" since it may apply to any message type.

**Comment by AArnott** [comment]
@milopezc, When [I asked](https://github.com/microsoft/vs-streamjsonrpc/pull/318#pullrequestreview-275257215) for you to rebase your changes to the v2.1 branch, it didn't require that you close your PR and open a new one. You can edit the target branch on an existing PR.
But more importantly, merely changing the target branch or opening a new PR with a different target branch is insufficient. You must *rebase* your commits to the v2.1 branch, which you evidently didn't do. Check the commits list in this PR and you'll see some commits unrelated to your change because they're still based on master.

You can fix this, without closing this PR, by following these steps on your machine:

```
git rebase --onto upstream/v2.1 8520f68~1 dev/milopezc/FixedTransmitExceptionHandling_959430 
git push -f
```

The commit ID that appears in that command line is *your* first commit related to this change. You're telling git to move your topic branch (last argument) to v2.1 (first argument), moving all commits up to and excluding the commit named. By stating your commit ID with `~1` at the end, that's git notation to identity the commit before it, so you're stating to consider the base of your topic branch to be just before your first commit, so that it moves your first commit.

That will take care of the rebasing. The next thing I would recommend is that you squash your commits since there are quite a few little fix-up commits. You can do that with these commands:

```
git merge upstream/v2.1
git reset --soft upstream/v2.1
git commit -m "Fixed exception handling in TransmitAsync"
git push -f
```

File: src/StreamJsonRpc/Resources.resx (hunk 1 above)
Comment (milopezc) [reply src/StreamJsonRpc/Resources.resx]: Fixed, I was unable to resolve this using CodeFlow for some reason, I'll just resolve it here.

Comment (AArnott) [reply src/StreamJsonRpc/Resources.resx]: CodeFlow "resolves" by editing the original comment, which github won't let you do unless it's your own comment.

**Pull request merged**